### DISCUSSION TITLE
Upgrading from v8 to v9: usage of require() in 'import-fresh' which is a dependency of eslintrc

### DISCUSSION THREAD

**Discussion by urzparat** [eslint/eslint#18820] [Q&A] [answered]
I've updated my config via the upgrade-script `npx @eslint/migrate-config .eslintrc.json`. Now I'm getting the following error:

> Error [ERR_REQUIRE_ESM]: require() of ES Module …/node_modules/eslint-config-xo/index.js from …/node_modules/import-fresh/index.js not supported.

I wonder why I can't find any issue for this problem, because `import-fresh` is a dependency of eslint. So how to fix this?

<details><summary>Used Versions</summary>
<pre>
    "devDependencies": {
        "@eslint/eslintrc": "^3.1.0",
        "@eslint/js": "^9.9.1",
        "@types/node": "^22.5.0",
        "eslint": "^9.9.1",
        "eslint-config-prettier": "^9.1.0",
        "eslint-config-xo": "^0.46.0",
        "eslint-config-xo-typescript": "^6.0.0",
        "eslint-plugin-prettier": "^5.2.1",
        "eslint-plugin-unicorn": "^55.0.0",
        "prettier": "^3.3.3",
        "prettier-eslint": "^16.3.0",
        "typescript": "^5.5.4"
    },
</pre>
</details>

**Comment by fasttime** [comment]
`eslint-config-xo` in version 0.46.0 does not export any legacy config that would work in a `.eslintrc.json` config file. Did you upgrade your dependencies after running `npx @eslint/migrate-config`? If not, it's possible that `eslint-config-xo` was being imported but not used. Consider showing your `.eslintrc.json` file so that we can have a look.

**Reply by urzparat** [reply]
Thank you for your answer! Yes, I upgraded all dependencies to the latest version (I hope).

The former `.eslintrc.json`:
```
{
    "plugins": [
        "prettier",
        "unicorn"
    ],
    "extends": [
        "xo",
        "xo-typescript",
        "prettier",
        "plugin:unicorn/recommended"
    ],
    "rules": {
        "prettier/prettier": [
            "warn",
            {
                "tabWidth": 4,
                "singleQuote": true,
                "trailingComma": "es5"
            }
        ],
        "unicorn/no-process-exit": 0,
        "unicorn/prefer-node-protocol": 0,
        "comma-dangle": ["error", "only-multiline"],
        "capitalized-comments": [
            "error",
            "always",
            {
                "line": {
                    "ignorePattern": ".*"
                },
                "block": {
                    "ignoreInlineComments": true,
                    "ignoreConsecutiveComments": true
                }
            }
        ],
        "@typescript-eslint/naming-convention": [
            "error",
            {
                "selector": "variable",
                "format": ["camelCase"]
            },
            {
                "selector": "variable",
                "modifiers": [
                    "destructured"
                ],
                "format": null
            },
            {
                "selector": "variable",
                "filter": {
                    "regex": "^API[_\\w]+",
                    "match": true
                },
                "format": ["UPPER_CASE"]
            }
        ]
    }
}
```

<details><summary>Which was converted to…</summary>
<pre>
import prettier from "eslint-plugin-prettier";
import unicorn from "eslint-plugin-unicorn";
import path from "node:path";
import { fileURLToPath } from "node:url";
import js from "@eslint/js";
import { FlatCompat } from "@eslint/eslintrc";

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const compat = new FlatCompat({
    baseDirectory: __dirname,
    recommendedConfig: js.configs.recommended,
    allConfig: js.configs.all
});

export default [{
    ignores: ["**/dist/", "**/node_modules/"],
}, ...compat.extends("xo", "xo-typescript", "prettier", "plugin:unicorn/recommended"), {
    plugins: {
        prettier,
        unicorn,
    },

    rules: {
        "prettier/prettier": ["warn", {
            tabWidth: 4,
            singleQuote: true,
            trailingComma: "es5",
        }],

        "unicorn/no-process-exit": 0,
        "unicorn/prefer-node-protocol": 0,
        "comma-dangle": ["error", "only-multiline"],

        "capitalized-comments": ["error", "always", {
            line: {
                ignorePattern: ".*",
            },

            block: {
                ignoreInlineComments: true,
                ignoreConsecutiveComments: true,
            },
        }],

        "@typescript-eslint/naming-convention": ["error", {
            selector: "variable",
            format: ["camelCase"],
        }, {
            selector: "variable",
            modifiers: ["destructured"],
            format: null,
        }, {
            selector: "variable",

            filter: {
                regex: "^API[_\\w]+",
                match: true,
            },

            format: ["UPPER_CASE"],
        }],
    },
}];
</pre>
</details>

**Reply by fasttime** [reply, marked as answer]
Since `eslint-config-xo` and `eslint-config-xo-typescript` have been updated to the new config system, they no longer need `FlatCompat` support, so you could import them directly and include them in your config like this:

```diff
+ import xoConfig from "eslint-config-xo";
+ import xoTypeScriptConfig from "eslint-config-xo-typescript";
+
  export default [{
      ignores: ["**/dist/", "**/node_modules/"],
-  }, ...compat.extends("xo", "xo-typescript", "prettier", "plugin:unicorn/recommended"), {
+     ...xoConfig,
+     ...xoTypeScriptConfig,
+     ...compat.extends("prettier", "plugin:unicorn/recommended"), {
      plugins: {
          prettier,
          unicorn,
      },
```

As per [this issue](https://github.com/eslint/eslint/issues/18093), `eslint-config-prettier` also supports flat config, so you could try to adjust it as well to make it work without `FlatCompat`.
